tests/test_type.py

```diff
@@ -82,3 +82,29 @@ def some_field(self, my_input: SomeModelInput) -> str: ...
     }
     """
     assert textwrap.dedent(str(schema)) == textwrap.dedent(expected).strip()
+
+
+def test_optimizer_hints_on_type():
+    class OtherModel(models.Model):
+        name = models.CharField(max_length=255)
+
+    class SomeModel(models.Model):
+        name = models.CharField(max_length=255)
+        other = models.ForeignKey(OtherModel, on_delete=models.CASCADE)
+
+    @strawberry_django.type(
+        SomeModel,
+        only=["name", "other", "other_name"],
+        select_related=["other"],
+        prefetch_related=["other"],
+        annotate={"other_name": models.F("other__name")},
+    )
+    class SomeModelType:
+        name: str
+
+    store = SomeModelType.__strawberry_django_definition__.store
```

Comment:
nit

obs. you need to `from strawberry_django.utils.typing import get_django_definition` as well

```suggestion
    store = get_django_definition(SomeModelType, strict=True), store
```